src/d3a/models/strategy/area_agents/two_sided_revenue_fee.py

```diff
@@ -0,0 +1,96 @@
+
+
+def update_incoming_bid_with_fee(source_bid, original_bid, tax_percentage=None):
+    if source_bid is None:
+        return original_bid
+    return source_bid
+
+
+def update_incoming_offer_with_fee(source_offer_price, original_offer_price, tax_percentage):
+    if source_offer_price is None:
+        return original_offer_price * (1 + tax_percentage)
+    return source_offer_price + original_offer_price * tax_percentage
+
+
+# Protocol B
+def calculate_fee_revenue_from_clearing_trade(
+        bid_propagated, bid_original,
+        offer_propagated, offer_original,
+        trade_rate_source, tax_percentage_n
+):
+    demand_side_tax = 0 if bid_original == 0 else 1 - bid_propagated / bid_original
```

Comment:
I know, you tried to follow the naming convention introduced in the ticket, but still, I am voting to attach the type of the variable. So, instead of `bid_propagated` I ask to rename to `bid_propagated_rate` .
Or do you think this will clatter the code?

Reply:
It will clutter it a bit, but it is good to have info about the unit of the variable. Fixing